tools/initMigrationToPP/initMigrationToPP.ts

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable no-await-in-loop, no-use-before-define, no-lonely-if */
+/* eslint-disable no-console, no-inner-declarations, no-undef, import/no-unresolved, import/extensions */
+import path = require('path');
+import fs = require('fs');
+
+import * as dotenv from 'dotenv';
+import { ethers } from 'hardhat';
+import { PolygonRollupManager } from '../../typechain-types';
+import { transactionTypes, genOperation } from '../utils';
+import '../../deployment/helpers/utils';
+import initMigrationToPPParams from './initMigrationToPP.json';
+import { checkParams, getDeployerFromParameters, getProviderAdjustingMultiplierGas } from '../../src/utils';
+import { logger } from '../../src/logger';
+import { decodeScheduleData } from '../../upgrade/utils';
+
+dotenv.config({ path: path.resolve(__dirname, '../../.env') });
+
+const dateStr = new Date().toISOString();
+const pathOutputJson = path.join(__dirname, `./initMigrationToPP-${dateStr}.json`);
+
+async function main() {
+    /*
+     * Check parameters
+     * Check that every necessary parameter is fullfilled
+     */
+    const mandatoryParameters = ['type', 'rollupID', 'newRollupTypeID', 'polygonRollupManagerAddress'];
+
+    // check create rollup type
+    switch (initMigrationToPPParams.type) {
+        case transactionTypes.EOA:
+        case transactionTypes.MULTISIG:
+            break;
+        case transactionTypes.TIMELOCK:
+            mandatoryParameters.push('timelockDelay');
+            break;
+        default:
+            throw new Error(`Invalid type ${initMigrationToPPParams.type}`);
+    }
+
+    checkParams(initMigrationToPPParams, mandatoryParameters);
+
+    logger.info(`Starting script to update rollup from ${initMigrationToPPParams.type}`);
+
+    // Load provider
+    logger.info('Load provider');
+    const currentProvider = getProviderAdjustingMultiplierGas(initMigrationToPPParams, ethers);
+
+    // Load deployer
+    logger.info('Load deployer');
+    const deployer = await getDeployerFromParameters(currentProvider, initMigrationToPPParams, ethers);
+    logger.info('Using with: ', deployer.address);
```

Comment:
I am not 100% sure but I think that for `logger.info` you have to format string like 
````
logger.info(`deploying implementation with: ${deployer.address}`)
````
to print variables